projects/portal/src/app/views/dialogs/wallets/ununifi/ununifi-import-wallet-with-private-key-form-dialog/ununifi-import-wallet-with-private-key-form-dialog.component.ts

```diff
@@ -0,0 +1,138 @@
+import { Clipboard } from '@angular/cdk/clipboard';
+import { Component, OnInit } from '@angular/core';
+import { MatDialogRef } from '@angular/material/dialog';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { cosmosclient } from '@cosmos-client/core';
+import { KeyType } from 'projects/portal/src/app/models/keys/key.model';
+import { StoredWallet, WalletType } from 'projects/portal/src/app/models/wallets/wallet.model';
+import { WalletService } from 'projects/portal/src/app/models/wallets/wallet.service';
+import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { createCosmosPrivateKeyFromString } from 'projects/portal/src/app/utils/key';
+
+@Component({
+  selector: 'view-ununifi-import-wallet-with-private-key-form-dialog',
+  templateUrl: './ununifi-import-wallet-with-private-key-form-dialog.component.html',
+  styleUrls: ['./ununifi-import-wallet-with-private-key-form-dialog.component.css'],
+})
+export class UnunifiImportWalletWithPrivateKeyFormDialogComponent implements OnInit {
+  isPasswordVisible: boolean = false;
+  privateWallet$: Observable<(StoredWallet & { mnemonic: string; privateKey: string }) | undefined>;
+  idSubject$: BehaviorSubject<string>;
+  id$: Observable<string>;
+  privateKeySubject$: BehaviorSubject<string>;
+  privateKey$: Observable<string>;
+  wallets$: Observable<StoredWallet[] | null | undefined>;
+
+  constructor(
+    private readonly dialogRef: MatDialogRef<UnunifiImportWalletWithPrivateKeyFormDialogComponent>,
+    private clipboard: Clipboard,
+    private readonly snackBar: MatSnackBar,
+    private walletService: WalletService,
+  ) {
+    this.wallets$ = this.walletService.storedWallets$;
+    this.idSubject$ = new BehaviorSubject<string>('');
+    this.idSubject$.next('');
+    this.id$ = this.idSubject$.asObservable();
+    this.privateKeySubject$ = new BehaviorSubject<string>('');
+    this.privateKeySubject$.next('');
+    this.privateKey$ = this.privateKeySubject$.asObservable();
+    this.privateWallet$ = combineLatest([this.id$, this.privateKey$]).pipe(
+      map(([id, privateKey]) => {
+        if (!privateKey) {
+          return {
+            id,
+            type: WalletType.ununifi,
+            mnemonic: '',
+            key_type: KeyType.secp256k1,
+            privateKey: '',
+            public_key: '',
+            address: '',
+          };
+        }
+
+        try {
+          const cosmosPrivateKey = createCosmosPrivateKeyFromString(
+            KeyType.secp256k1,
+            privateKey,
+          );
+          if (!cosmosPrivateKey) {
```

Comment:
If you enter the publickey instead of the privatekey, a 33-byte privatekey will be returned and pass this check.
So, I added a check to `projects/portal/src/app/utils/key.ts ` to confirm that Uint8Array is 32 bytes.